tests/lib/devicetree/api/app.overlay

```diff
@@ -547,5 +547,25 @@
 			      "token_second_idx_two",
 			      "token_second_idx_three";
 		};
+
+		test_str_float_0: string-float-0 {
+			compatible = "vnd,string-float";
+			val = "0.1234";
+		};
+
+		test_str_float_1: string-float-1 {
+			compatible = "vnd,string-float";
+			val = "0.9e-3";
+		};
+
+		test_str_array_float_0: string-array-float-0 {
+			compatible = "vnd,string-array-float";
+			val = "1.0e2", "2.0e2", "3.0e2", "4.0e2";
+		};
+
+		test_str_array_float_1: string-array_float-1 {
+			compatible = "vnd,string-array-float";
+			val = "0.01", "0.1", "1.0", "10.0";
```

Comment:
Can you add a test that relies on multiple tokens appearing in the resulting macro expansion too, to make sure that's covered?

Reply:
I was wondering how to do that - how to detect on C level that we have A B C tokens. I had to define a macros: A as value, B as operation and C as another value and then test if it gives the expected result. Please check if this approach looks good to you.